src/App.tsx

```diff
@@ -14,11 +15,74 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGoods: string;
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    selectedGoods: 'Jam',
+  };
+
+  selectedGood = (item: string) => {
+    this.setState({ selectedGoods: item });
+  };
+
+  clearItem = () => {
+    this.setState({ selectedGoods: '' });
+  };
+
+  render() {
+    const { selectedGoods } = this.state;
+
+    return (
+      <div className="App">
+        <div className="App__top">
+          {selectedGoods
+            ? <h1>{`Selected good: - ${selectedGoods}`}</h1>
+            : <h1>No goods selected!</h1>}
+
+          {selectedGoods && (
+            <button
+              type="button"
+              className="App__button--clear button"
+              onClick={() => this.clearItem()}
```

Comment:
якщо метод не приймає парамертів то можна передавати посилання на метод без його виклику
```suggestion
              onClick={this.clearItem}
```